csatar-plugins/forms/components/BasicForm.php

```diff
@@ -106,7 +108,8 @@ private function getForm() {
             $this->form_id = $form->id;
             return $form;
         } else {
-            // TODO: exception
+            $error = e(trans('csatar.forms::lang.errors.formNotFound'));
+            throw new ApplicationException($error);
```

Comment:
Az `ApplicationException` túl általános. Vagy egy már létező exceptiont kéne használni (pl. `NotFoundException`) vagy definiálni egy újat: `FormNotFoundException`.

Lásd pl [itt](https://stackoverflow.com/questions/11467045/how-to-name-exceptions-php):
```
In the past I had trouble with using one exception for different errors
because when I catched the exceptions, I sometimes catch exception
that I did not want to catch.
```
Ez itt elég jó válasznak tűnik:
```
It's good to have a tree of exceptions, extending from the most general to the most specific, for instance, your StatementAlreadyCommentedException (That's how it should be named, with the Exception part at the end) should inherit from InvalidUserInputException, or maybe AlreadyExistsException exception, and those should inherit from something more general, like Exception.

This way, you can catch all exceptions more specific than the one you're searching (should you ever need it). (You can catch Exception, and it would catch every possible exception, you can catch AlreadyExistsException and it would catch any exception extending it.
```

Reply:
Ez a hiba akkor jelenik meg ha a komponensbe nem állított be "Form"-ot az aki betette a komponenst. Nem lesz ilyen gondunk.